File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.go
func minSubArrayLen(target int, nums []int) int {
    result := len(nums) + 1
    left := 0
    currentTotal := 0

    for right := 0; right < len(nums); right++ {
        currentTotal += nums[right]

        for currentTotal >= target {
            result = min(result, right - left + 1)
            currentTotal -= nums[left];
            left++
        } 
    }
    if result == len(nums) + 1 {
        return 0
    }
    return result
}

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int left = 0;
        int right = 0;
        int currentSum = 0;
        int result = Integer.MAX_VALUE;

        while(right < nums.length)
        {
            currentSum += nums[right];

            while(currentSum >= target && left <= right)
            {
                result = Math.min(result, right - left + 1);
                currentSum -= nums[left];
                left++;
            }
            right++;
        }
        return result == Integer.MAX_VALUE ? 0 : result;
    }
}